<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:arg name="mid360" default="true"/>
  <xacro:include filename="base.urdf.xacro"/>
  <xacro:include filename="left_wheel.urdf.xacro"/>
  <xacro:include filename="right_wheel.urdf.xacro"/>
  <xacro:include filename="left_caster.urdf.xacro"/>
  <xacro:include filename="right_caster.urdf.xacro"/>
  <xacro:include filename="mid360.urdf.xacro"/>
  <xacro:include filename="imu.urdf.xacro"/>

  <!-- <link name="world"> -->
  <!-- </link> -->
  <!---->
  <!-- <joint name="world_joint" type="fixed"> -->
  <!--   <parent link="world"/> -->
  <!--   <child link="base_link"/> -->
  <!--   <origin xyz="0 0 0.01"/> -->
  <!-- </joint> -->

  <link name="base_link"/>

  <joint name="base_link_to_base_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base"/>
    <origin xyz="0 0 0.3"/>
  </joint>

  <link name="base"/>

  <xacro:base name="body" parent="base">
    <origin xyz="0.0569 -0.0028 0.2234" rpy="0 0 0"/>
  </xacro:base>

  <!-- <joint name="odom_joint" type="fixed"> -->
  <!--   <parent link="base"/> -->
  <!--   <child link="odom"/> -->
  <!--   <origin xyz="0 0 0" rpy="0 0 0"/> -->
  <!-- </joint> -->
  <!---->
  <!-- <link name="odom"/> -->


  <xacro:left_wheel name="left_wheel" parent="body">
    <origin xyz="-0.0569 0.3191 -0.2234" rpy="0 0 0"/>
  </xacro:left_wheel>

  <xacro:right_wheel name="right_wheel" parent="body">
    <origin xyz="-0.0569 -0.3135 -0.2234" rpy="0 0 0"/>
  </xacro:right_wheel>

  <xacro:left_caster name="left_caster" parent="body">
    <origin xyz="0.4024 0.3006 -0.3152" rpy="0 0 0"/>
  </xacro:left_caster>

  <xacro:right_caster name="right_caster" parent="body">
    <origin xyz="0.3933 -0.295 -0.3152" rpy="0 0 0"/>
  </xacro:right_caster>

  <xacro:mid360 name="livox_frame" parent="body">
    <origin xyz="-0.1201 0.0026 0.9655" rpy="${pi} 0 ${pi/2}"/>
  </xacro:mid360>

  <xacro:imu_sensor name="livox_imu" parent="livox_frame" topic="mid360_imu">
    <origin xyz="0.0 0.0 0.0" rpy="0 0 0"/>
  </xacro:imu_sensor>
</robot>
